### PR TITLE
Add in Rényi divergences

### PR DESCRIPTION
**Pull request by richardreeve**
Added in Rényi divergences, as discussed in #45 and JuliaStats/StatsBase.jl#131. Goes with Rényi entropy pull for JuliaStats/StatsBase.jl#190 repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,6 +139,7 @@ Each distance corresponds to a distance type. The type name and the correspondin
 |  CorrDist            |  corr_dist(x, y)         | cosine_dist(x - mean(x), y - mean(y)) |
 |  ChiSqDist           |  chisq_dist(x, y)        | sum((x - y).^2 / (x + y)) |
 |  KLDivergence        |  kl_divergence(x, y)     | sum(p .* log(p ./ q)) |
+| RenyiDivergence   | renyi_divergence(x, y, k) | log(sum( x .* (x ./ y) .^ (k - 1))) / (k - 1) |
```
Comment (ararslan) [inline README.md]: Can you make the `|`s line up? Not that it actually matters, it's just bugging me. 😛

File: src/metrics.jl [hunk 2]
```diff
@@ -141,6 +163,28 @@ chisq_dist(a::AbstractArray, b::AbstractArray) = evaluate(ChiSqDist(), a, b)
...
+
+@inline eval_op(dist::RenyiDivergence, ai, bi) = ai, (ai == 0 ? zero(ai) :
+                                                      (dist.is_normal ?
+                                                       ai .* ((ai ./ bi) .^ dist.p) :
+                                                       (dist.is_zero ?
+                                                        bi :
+                                                        (dist.is_one ?
+                                                         ai * log(ai / bi) :
+                                                         # otherwise α = ∞
+                                                         ai / bi))))
```
Comment (ararslan) [inline src/metrics.jl]: This is pretty hard to read. Would you mind putting it into the long form function syntax with `if` statements in place of ternaries? E.g.
```julia
@inline function eval_op(dist::RenyiDivergence, ai, bi)
    if ai == 0
        zero(ai)
    ...
end
```

**Comment by richardreeve** [comment]
@ararslan These are both addressed now, but the checks will all fail because of JuliaStats/StatsBase.jl#192.

**Comment by richardreeve** [comment]
@ararslan Okay, sorry for all of fiddling, but the tests now increase coverage except in places which have nothing to do with me - I don't know why they now show as not covered, but they clearly are when you look at the coveralls transcript (and they were correctly identified before).

**Comment by ararslan** [comment]
Coverage can be a little weird sometimes. Don't sweat a decrease of 0.3%. :wink:

**Comment by ararslan** [comment]
I don't know much about Rényi divergences so I don't think I'm the best person to perform a thorough code review here. The code looks fine to me, but it'd be great if someone here who knows more about this stuff could chime in.

**Comment by richardreeve** [comment]
I appreciate this is only a partial solution, but there is reasonably extensive testing of the code. Otherwise it may be years before a developer who uses Rényi entropies comes along to check it. For what it's worth, the relevant bit of Renyi's paper is attached, though it doesn't deal with the cases alpha == 0 or alpha == Inf.
renyi

**Comment by richardreeve** [comment]
Note that for historical reasons he uses base 2 for logs, whereas we use natural logs (naturally!).

File: src/metrics.jl [hunk 3]
```diff
@@ -141,6 +163,53 @@ chisq_dist(a::AbstractArray, b::AbstractArray) = evaluate(ChiSqDist(), a, b)
...
+    elseif dist.is_zero
+        return ai, bi
+    elseif dist.is_one
+        return ai, ai * log(ai / bi)
+    else # otherwise q = ∞
+        return ai, ai / bi
+    end
+end
+
+@inline function eval_reduce{T<:AbstractFloat}(dist::RenyiDivergence,
```
Comment (KristofferC) [inline src/metrics.jl]: Isnt this function type unstable? In some cases you return a `::T` and in some caes `(::T, ::T)`?

**Pull request merged**